### PR TITLE
WIP 0.3.0

### PR DESCRIPTION
**Pull request by crisptrutski**
@Deraen Let's use this as an integration branch to PR against for any features/fixes to include in the next breaking release.

I you have the patience (or already did the work), would also take same contributions against `master` and would cut PATCH releases for them. 

But would prefer not to port extra stuff over now, so favour here :smile:

### CODE REVIEW

**Comment by Deraen** [comment]
I can rework my changes on top of this. I guess the some of the changes I did are already fixed by this. I'll try to find time to check this on weekend.

If this is ready, I think it would be easiest to just merge this to master and I'll make my PRs against that.

**Comment by crisptrutski** [comment]
This could still use some polish even on what I've covered already (for example printing warnings for removed arguments), so will leave it open for bit.

Thanks for chipping in and for all the guidance so far - if I have time may port the work you've already shared, and implement what I can of the other issues you've opened.

File: src/crisptrutski/boot_cljs_test.clj [hunk 1]
```diff
@@ -27,145 +23,171 @@
...
+        edn-file   (io/file tmp-main (str out-main ".edn"))
+        src-path   (relative src-file)
+        edn-path   (relative edn-file)
+        exists?    (into #{} (map core/tmp-path) (u/cljs-files fileset))
+        suite?     (or (exists? src-path) (exists? (str/replace src-path ".cljs" ".cljc")))
+        edn?       (exists? edn-path)
+        edn        (when edn? (read-string (slurp (core/tmp-file (core/tmp-get fileset edn-path)))))
+        ;; subset namespaces to those directly required by .cljs.edn
+        ;; TODO: better would be to use transitive requires, once we have the dep graph
+        namespaces (if edn (filter (set (:require edn)) namespaces) namespaces)
```
Comment (crisptrutski) [inline src/crisptrutski/boot_cljs_test.clj:66]: @Deraen one part I'm not super happy about, but better than errors trying to run a namespace that hasn't loaded

File: src/crisptrutski/boot_cljs_test.clj [hunk 2]
```diff
@@ -27,145 +23,171 @@
...
+      (info "Using %s...\n" src-path)
+      (do
+        (info "Writing %s...\n" src-path)
+        (spit src-file (gen-suite-ns suite-ns namespaces))))
+    (if edn?
+      ;; ensure that .cljs file is required by .cljs.edn, if it's being created.
+      (if suite?
+        (info "Using %s...\n" edn-path)
+        (do (info "Updating %s...\n" edn-path)
+            (spit edn-file (update edn :require (fn [xs] (when-not (some #{suite-ns} xs)
```
Comment (crisptrutski) [inline src/crisptrutski/boot_cljs_test.clj:79]: A similar wart, let me know if you have any immediate thoughts on these niggles

**Pull request merged**